tests/common/mod.rs

```diff
@@ -287,10 +319,86 @@ where
 	}
 }
 
+pub fn send_request_enc<OUT>(
+	sec_req_id: u32,
+	internal_request_id: u32,
+	dest: &str,
+	req: &str,
+	shared_key: &SecretKey,
+) -> Result<Result<OUT, WalletAPIReturnError>, api::Error>
+where
+	OUT: DeserializeOwned,
+{
+	let url = Url::parse(dest).unwrap();
```

Comment:
Should we really have all of these unwraps in `send_request_enc` and `derive_ecdh_key`? What RPC error code would be returned if these panic?

Reply:
It's just test helper code, I don't usually worry too much about unwraps there. I've been careful to ensure the server side returns proper errors for all conditions (and never unwraps)